### DISCUSSION TITLE
Welcome to nibabel Discussions!

### DISCUSSION THREAD

**Discussion by effigies** [nipy/nibabel#1215] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

  This is an experiment, and we're not sure how it's going to go. From observing other projects, some have made discussions work for them, and others have found other discussion platforms work better. Other places you can look for help include the [neuroimaging @ python mailing list](https://mail.python.org/mailman/listinfo/neuroimaging) or the [NeuroStars Discourse forum](https://neurostars.org/). For bug reports, we recommend opening [issues](https://github.com/nipy/nibabel/issues/new) (although you should first [search existing issues](https://github.com/nipy/nibabel/issues) to see if yours has already been addressed).

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ZviBaratz** [comment]
Hello 👋🏼 
I'm Zvi, I've been using NiBabel for a few years now, mostly to read and write NIfTI files. 
I believe NiBabel has a pivotal role in the Python neuroimaging community—not only as a software tool, but also in terms of it being a common resource which encourages the interaction of users from different domains. 
Thank you, lead maintainers, for all your great work 🙏🏼

**Comment by romainVala** [comment]
Hello
I am a french research engineer working on processing (brain) MRI data since 20 years. It tooks me time to get rid of my matlab code I started with, but now I started new project I just enjoy the python community (and nibable !!!). thank for proposing this discussion, here are 3 topics I would be interested to discuss  (may be in separate threads ?)

- why 2 orientation in the nifti header ? :
When I started (my pdh) we only had the analyse format.  This was a "bad time" for instance when we had a sagittal T1w and axial EPI, it required a manual flip of the anatomical image before trying a co-registration. 
Then the nifti initiative came in and define the nifti format. Finally the volume information was stored in the image ! Good move, expect that they define 2 orientations matrix, without a clear definition on which one to use ! (and the constrain to stay compatible with analyze format was a bad constrain)
since then I am often come to silent (but important) bugs when playing with different software, because of those 2 orientations defined in the header. (on soft will use only the qform and set the sform to zero for output volumes, and an other soft will do the opposite .. leading to the loss of orientation information ) 


- surface data format : I am not used to work with suface data but when I try to process / visualize surface I always need to solve conversion issues and often I have difficulties with surface mm orientation so that it matches a given nifti volume)

- quaternion or euler angle. (dual quaternion or matrix representation for rigid motion) :  I recently had to consider how to average several rotations. This is a difficult problem and while reading about it, I discover the dual quaternion representation. It has been widely adopted in the gaming community mainly because it allows correct interpolation. It is still not clear to me whether we really need it or if the 4x4 affine matrix is just fine. (Note that for the gaming community euler representaiton is realy the worst choice one can made ! We understand why when one try to convert a given rigid motion from on software to an other (fsl / ants / nifti reg / scipy ... they all used different conventions and reverse engineering is painful ...  Note sure if we need to go that fare but just for the mathematical beauty I highly recommend the bivector forum (https://bivector.net/ and in the doc section SIBGRAPI 2021 introduction on PGA)  Steven De Keninck is making very nice presentation to explain how dual quaternion representation is deduced from geometric algebra (Mathematical group theory can be useful from a practical/engineering point of view ... amazing isn't it ?)

**Reply by effigies** [reply]
Yes, I would open new threads for these.

**Reply by ZviBaratz** [reply]
Hi Romain,
Congratulations on making the move from MATLAB – must take some determination after using it for so long.
These definitely seem like great topics for independent discussion threads!
